pandas/tests/groupby/test_groupby_subclass.py

```diff
@@ -103,3 +103,89 @@ def test_groupby_resample_preserves_subclass(obj):
     # Confirm groupby.resample() preserves dataframe type
     result = df.groupby("Buyer").resample("5D").sum()
     assert isinstance(result, obj)
+
+
+def test_groupby_overridden_methods():
+    class UnitSeries(Series):
+        @property
+        def _constructor(self):
+            return UnitSeries
+
+        @property
+        def _constructor_expanddim(self):
+            return UnitDataFrame
+
+        def mean(self, *args, **kwargs):
+            return 1
+
+        def median(self, *args, **kwargs):
+            return 2
+
+        def std(self, *args, **kwargs):
+            return 3
+
+        def var(self, *args, **kwargs):
+            return 4
+
+        def sem(self, *args, **kwargs):
+            return 5
+
+        def prod(self, *args, **kwargs):
+            return 6
+
+        def min(self, *args, **kwargs):
+            return 7
+
+        def max(self, *args, **kwargs):
+            return 8
+
+    class UnitDataFrame(DataFrame):
+        @property
+        def _constructor(self):
+            return UnitDataFrame
+
+        @property
+        def _constructor_expanddim(self):
+            return UnitSeries
+
+        def mean(self, *args, **kwargs):
+            print("UnitDataFrame mean")
+            return 1
+
+        def median(self, *args, **kwargs):
+            return 2
+
+        def std(self, *args, **kwargs):
+            return 3
+
+        def var(self, *args, **kwargs):
+            return 4
+
+        def sem(self, *args, **kwargs):
+            return 5
+
+        def prod(self, *args, **kwargs):
+            return 6
+
+        def min(self, *args, **kwargs):
+            return 7
+
+        def max(self, *args, **kwargs):
+            return 8
+
+    params = ["a", "b"]
+    data = np.random.rand(4, 2)
+    udf = UnitDataFrame(data, columns=params)
+    udf["group"] = np.ones(4, dtype=int)
+    udf.loc[2:, "group"] = 2
+
+    assert udf.mean() == 1
+    assert all(udf.groupby("group").mean() == 1)
+    assert all(udf.groupby("group").median() == 1)
+    assert all(udf.groupby("group").std() == 1)
+    assert all(udf.groupby("group").var() == 1)
+    assert all(udf.groupby("group").sem() == 1)
+    assert all(udf.groupby("group").prod() == 1)
+    assert all(udf.groupby("group").min() == 1)
+    assert all(udf.groupby("group").max() == 1)
+    # print(udf.groupby('group').beans()) # AttributeError
```

Comment:
* Shouldn't it be 2, 3, 4, etc. for median, std, var, etc.?
* What about methods that don't appear in `pandas/core/groupby/groupby.py`, e.g.`skew()` or `kurt()`? Might be good to include some of those in the test, too?

Reply:
> * Shouldn't it be 2, 3, 4, etc. for median, std, var, etc.?

I'd already fixed this! I hadn't realised as I should also have been using `np.all`

> * What about methods that don't appear in `pandas/core/groupby/groupby.py`, e.g.`skew()` or `kurt()`? Might be good to include some of those in the test, too?